code/modules/mining/aux_base_camera.dm

```diff
@@ -276,4 +276,246 @@ datum/action/innate/aux_base/install_turret/Activate()
+	eyeobj = new /mob/camera/aiEye/remote/arena_construction(get_turf(src))
+	eyeobj.origin = src
+
+/obj/machinery/computer/camera_advanced/arena_construction/GrantActions(mob/living/user)
+	..()
+	if(V)
+		V.target = src
+		V.Grant(user)
+		actions += V
+	if(I)
+		I.target = src
+		I.Grant(user)
+		actions += I
+	if(L)
+		L.target = src
+		L.Grant(user)
+		actions += L
+	if(S)
+		S.target = src
+		S.Grant(user)
+		actions += S
+	if(Z)
+		Z.target = src
+		Z.Grant(user)
+		actions += Z
+	if(D)
+		D.target = src
+		D.Grant(user)
+		actions += D
+	if(F)
+		F.target = src
+		F.Grant(user)
+		actions += F
+	if(N)
+		N.target = src
+		N.Grant(user)
+		actions += N
+	if(E)
+		E.target = src
+		E.Grant(user)
+		actions += E
+	if(H)
+		H.target = src
+		H.Grant(user)
+		actions += H
+	eyeobj.invisibility = 0
+
+/obj/machinery/computer/camera_advanced/arena_construction/remove_eye_control(mob/living/user)
+	..()
+	eyeobj.invisibility = INVISIBILITY_MAXIMUM //Hide the eye when not in use.
+
+/datum/action/innate/arena_base//Parent for arena construction actions
+	icon_icon = 'icons/mob/actions/actions_construction.dmi'
+	var/mob/living/C //Mob using the action
+	var/mob/camera/aiEye/remote/base_construction/remote_eye //Console's eye mob
+	var/obj/machinery/computer/camera_advanced/arena_construction/comp //Console itself
+
+/datum/action/innate/arena_base/Activate()
+	if(!target)
+		return FALSE
+	C = owner
+	remote_eye = C.remote_control
+	comp = target
+	if(!check_spot())
+		return FALSE
+	if(hazard_limited())
+		return FALSE
+	return TRUE
+
+/datum/action/innate/arena_base/proc/check_spot()
+	var/turf/build_target = get_turf(remote_eye)
+	var/area/build_area = get_area(build_target)
+	if(!istype(build_area, /area/maintenance/arena))
+		to_chat(owner, "<span class='warning'>You can only build within the arena!</span>")
+		return FALSE
+	return TRUE
+
+/datum/action/innate/arena_base/proc/hazard_limited()
+	for(var/obj/O in comp.hazards)
+		if(QDELETED(O))
+			comp.hazards -= O
+	if(LAZYLEN(comp.hazards) >= 12)
+		to_chat(owner, "<span class='warning'>You have reached the arena's hazard limit - clear the previous hazards to create more!</span>")
+		return TRUE
+	return FALSE
+
+
+datum/action/innate/arena_base/turfchange
+	name = "Place Default Turf"
+	button_icon_state = "build"
+	var/turf_type = /turf/open/floor/plasteel/showroomfloor
+
+datum/action/innate/arena_base/turfchange/Activate()
+	. = ..()
+	if(.)
+		var/turf/T = get_turf(remote_eye)
+		var/turf/open/Newt = T.ChangeTurf(turf_type, null, CHANGETURF_IGNORE_AIR)
+		Newt.planetary_atmos = 0
+		playsound(Newt, 'sound/machines/click.ogg', 50, 1)
+		new /obj/effect/temp_visual/small_smoke(Newt)
+		comp.hazards += Newt
+
+datum/action/innate/arena_base/turfchange/lava
+	name = "Place Lava"
+	icon_icon = 'icons/turf/floors.dmi'
+	button_icon_state = "lava"
+	turf_type = /turf/open/lava/smooth
+
+datum/action/innate/arena_base/turfchange/ice
+	name = "Place Lava"
```

Comment:
The trap of copy-pasting is that you forget to change certain variables! This should obviously be "Place Ice" or something.

Reply:
Thanks for pointing that out. I've managed to successfully notice and ignore that issue 3 times now.